workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-etc-cron/src/main/java/com/liferay/marketplace/MarketplaceCommandLineRunner.java

```diff
@@ -0,0 +1,203 @@
+
+import org.json.JSONObject;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.CommandLineRunner;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.MediaType;
+import org.springframework.stereotype.Component;
+import org.springframework.web.reactive.function.client.WebClient;
+
+/**
+ * @author Wellington Barbosa
+ */
+@Component
+public class MarketplaceCommandLineRunner implements CommandLineRunner {
+
+	public void run(String... args) throws Exception {
+		_processExpiredTrials();
+		_processOnHoldTrials();
+	}
+
+	private void _deleteTrial(long orderId) throws Exception {
+		WebClient.create(
+			_marketplaceSpringBootUrl
+		).delete(
+		).uri(
+			"/trial/" + orderId
+		).header(
+			HttpHeaders.AUTHORIZATION,
+			_liferayOAuth2AccessTokenManager.getAuthorization(
+				"liferay-marketplace-etc-cron-oauth-application-headless-" +
+					"server")
+		).accept(
+			MediaType.APPLICATION_JSON
+		).retrieve(
+		).bodyToMono(
+			String.class
+		).block();
+	}
+
+	private JSONObject _getAvailabilityJSONObject() throws Exception {
+		return new JSONObject(
+			WebClient.create(
+				_marketplaceSpringBootUrl
+			).get(
+			).uri(
+				"/trial/availability"
+			).header(
+				HttpHeaders.AUTHORIZATION,
+				_liferayOAuth2AccessTokenManager.getAuthorization(
+					"liferay-marketplace-etc-cron-oauth-application-headless-" +
+						"server")
+			).accept(
+				MediaType.APPLICATION_JSON
+			).retrieve(
+			).bodyToMono(
+				String.class
+			).block());
+	}
+
+	private OrderResource _getOrderResource() throws Exception {
+		URL liferayDXPURL = new URL(
+			_lxcDXPServerProtocol + "://" + _lxcDXPMainDomain);
+
+		return OrderResource.builder(
+		).endpoint(
+			liferayDXPURL
+		).header(
+			HttpHeaders.AUTHORIZATION,
+			_liferayOAuth2AccessTokenManager.getAuthorization(
+				"liferay-marketplace-etc-cron-oauth-application-headless-" +
+					"server")
+		).build();
+	}
+
+	private void _postTrial(Order order) throws Exception {
+		WebClient.create(
+			_marketplaceSpringBootUrl
+		).post(
+		).uri(
+			"/trial/provisioning"
+		).header(
+			HttpHeaders.AUTHORIZATION,
+			_liferayOAuth2AccessTokenManager.getAuthorization(
+				"liferay-marketplace-etc-cron-oauth-application-headless-" +
+					"server")
+		).accept(
+			MediaType.APPLICATION_JSON
+		).contentType(
+			MediaType.APPLICATION_JSON
+		).bodyValue(
+			new JSONObject(
+			).put(
+				"classPK", order.getId()
+			).put(
+				"modelDTOOrder",
+				new JSONObject(
+				).put(
+					"accountId",
+					order.getAccountId(
+					).toString()
+				)
+			).toString()
+		).retrieve(
+		).bodyToMono(
+			String.class
+		).block();
+	}
+
+	private void _processExpiredTrials() throws Exception {
+		Page<Order> ordersPage = _getOrderResource().getOrdersPage(
+			"", "orderStatus/any(x:(x eq " + _ORDER_STATUS_COMPLETED + "))",
```

Comment:
Hey @welitonpe I forgot to comment with you about this filter, we should filter by the orderType related to trials, in this case "SOLUTIONS7".

Otherwise we might be calling the API with all the orders completed (dangerous for us.)

```suggestion
			"", "orderStatus/any(x:(x eq " + _ORDER_STATUS_COMPLETED + ")) and orderTypeExternalReferenceCode eq 'SOLUTIONS7'",
```